llvm/lib/Transforms/Vectorize/VPlanCFG.h

```diff
@@ -306,6 +306,24 @@ template <> struct GraphTraits<VPlan *> {
   }
 };
 
+inline bool VPlan::isExitBlock(VPBlockBase *VPBB) {
+  if (!isa<VPIRBasicBlock>(VPBB) || VPBB->getNumSuccessors() ||
```

Comment:
Could we just check if it is one of the exit blocks returned by `getExitBlocks`? Might be slightly slower, but shouldn't matter as it is only called at most once per VPBasicBlock.

Of it that is not possible assert that it only returns true if the block is in `getExitBlocks`, to make sure they don't disagree.

Does this need to be defined in `llvm/lib/Transforms/Vectorize/VPlanCFG.h` or could it be moved to VPlan.cpp?

Reply:
It should be fine to define `isExitBlock` in `VPlan.cpp`, which includes `VPlanCFG.h`?

Reply:
Sorry I don't know what's going on here. Not sure if it's my fault that it's editing your comment, or something weird in github?